drivers/sensor/ite_vcmp_it8xxx2/vcmp_ite_it8xxx2.c

```diff
@@ -0,0 +1,403 @@
+			ret = -ENOTSUP;
+			break;
+		}
+
+		if (val->val2 != 0) {
+			LOG_ERR("Vcmp%d threshold not support fractional part", config->vcmp_ch);
+			ret = -ENOTSUP;
+			break;
+		}
+
+		/* Set threshold raw value */
+		*reg_vcmpxthrdatl = (uint8_t)(val->val1 & 0xff);
+		*reg_vcmpxthrdatm = (uint8_t)((val->val1 >> 8) & 0xff);
+
+		/* Set lower or higher threshold */
+		if (attr == SENSOR_ATTR_UPPER_THRESH) {
+			*reg_vcmpxctl |= IT8XXX2_VCMP_GREATER_THRESHOLD;
+		} else {
+			*reg_vcmpxctl &= ~IT8XXX2_VCMP_GREATER_THRESHOLD;
+		}
+
+		break;
+	case SENSOR_ATTR_LOWER_VOLTAGE_THRESH:
+	case SENSOR_ATTR_UPPER_VOLTAGE_THRESH:
+		/*
+		 * Tranfrom threshold from mv to raw
+		 * NOTE: CMPXTHRDAT[9:0] = threshold(mv) * 1024 / 3000(mv)
+		 */
+		reg_val = (val->val1 * VCMP_RESOLUTION / VCMP_MAX_MVOLT);
+
+		if (reg_val > VCMP_RESOLUTION) {
+			LOG_ERR("Vcmp%d threshold only support 10-bits", config->vcmp_ch);
+			ret = -ENOTSUP;
+			break;
+		}
+
+		if (val->val2 != 0) {
+			LOG_ERR("Vcmp%d threshold not support fractional part", config->vcmp_ch);
+			ret = -ENOTSUP;
+			break;
+		}
+
+		/* Set threshold raw value */
+		*reg_vcmpxthrdatl = (uint8_t)(reg_val & 0xff);
+		*reg_vcmpxthrdatm = (uint8_t)((reg_val >> 8) & 0xff);
+
+		/* Set lower or higher threshold */
+		if (attr == (uint16_t) SENSOR_ATTR_UPPER_VOLTAGE_THRESH) {
+			*reg_vcmpxctl |= IT8XXX2_VCMP_GREATER_THRESHOLD;
+		} else {
+			*reg_vcmpxctl &= ~IT8XXX2_VCMP_GREATER_THRESHOLD;
+		}
+
+		break;
+	case SENSOR_ATTR_ALERT:
+		if (!!val->val1) {
+			clear_vcmp_status(dev, config->vcmp_ch);
+			vcmp_enable(dev, 1);
+		} else {
+			vcmp_enable(dev, 0);
+			clear_vcmp_status(dev, config->vcmp_ch);
+		}
+
+		break;
+	default:
+		ret = -ENOTSUP;
+	}
+
+	return ret;
+}
+
+static int vcmp_ite_it8xxx2_attr_get(const struct device *dev,
+				     enum sensor_channel chan,
+				     enum sensor_attribute attr,
+				     struct sensor_value *val)
+{
+	return -ENOTSUP;
+}
+
+static int vcmp_ite_it8xxx2_trigger_set(const struct device *dev,
+					const struct sensor_trigger *trig,
+					sensor_trigger_handler_t handler)
+{
+	const struct vcmp_it8xxx2_config *const config = dev->config;
+	struct vcmp_it8xxx2_data *const data = dev->data;
+
+	if (trig == NULL || handler == NULL) {
+		return -EINVAL;
+	}
+
+	if (trig->type != SENSOR_TRIG_THRESHOLD ||
+	    trig->chan != SENSOR_CHAN_VOLTAGE) {
+		return -ENOTSUP;
+	}
+
+	data->handler = handler;
+
+	vcmp_work_addr[config->vcmp_ch] = (uint32_t) &data->work;
+
+	return 0;
+}
+
+static int vcmp_it8xxx2_channel_get(const struct device *dev,
+				    enum sensor_channel chan,
+				    struct sensor_value *val)
+{
+	const struct vcmp_it8xxx2_config *const config = dev->config;
+
+	if (chan != SENSOR_CHAN_VOLTAGE) {
+		return -ENOTSUP;
+	}
+
+	if (val == NULL) {
+		return -EINVAL;
+	}
+
+	/*
+	 * It8xxx2 adc and comparator module read automatically, according to
+	 * {ADCCTS1, ADCCTS2} and VCMPSCP register setting.
+	 */
+	val->val1 = config->vcmp_ch;
```

Comment:
is this always returning a constant value?

Reply:
It returns the channel of current voltage comparator device, so the value will be 0~5.